src/libs/web3-data-provider/Web3Provider.tsx

```diff
@@ -432,6 +440,16 @@ export const Web3ContextProvider: React.FC<{ children: ReactElement }> = ({ chil
     setAccountLoading(loading);
   }, [loading]);
 
+  useEffect(() => {
+    if (account && walletType && walletName) {
+      trackEvent(AUTH.CONNECT_WALLET, {
+        walletType: walletType,
+        walletName: walletName,
+        walletAddress: account,
+      });
+    }
+  }, [account, walletType, walletName]);
+
```

Comment:
I think you could move the track event inside the connect wallet function, unless there is something im missing thats always the entry point for wallets